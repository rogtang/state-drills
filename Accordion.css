
.Accordion {
    margin: 0;
    padding: 0;
    list-style-type: none;
  }
  
  .accordion-item {
    margin-bottom: 2px;
  }
  
  .accordion-item button {
    background: lightgrey;
    border: 1px solid #222;
    padding: 10px 20px;
    width: 100%;
    font-size: 18px;
    font-weight: bold;
  }
  
  .accordion-item p {
    background: #fdfcfc;
    border: 1px solid #777;
    border-top: none;
    margin: 0;
    padding: 20px 10px;
  }