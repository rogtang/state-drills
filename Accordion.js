import React from 'react';
import './Accordion.css'

class Accordion extends React.Component {
    static defaultProps = { 
        sections: []
    };
    state = {
        activeSectionIndex: null,
    }
    renderItem(section, index, activeSectionIndex) {
        return (
            <li className="accordion-item" key={index}>
                <button type='button' onClick={() => this.handleActiveSection(index)}>
                    {section.title}
                </button>
                {(activeSectionIndex === index) &&<p>{section.content}</p>}
            </li>
        )
    }

    handleActiveSection = (index) => {
        this.setState({activeSectionIndex: index})
    }

    render() {
        return (
            <ul className="accordion">
                {this.props.sections.map((section, index) => 
                this.renderItem(section, index, this.state.activeSectionIndex))
                }
            </ul>
        )
    }
}

export default Accordion