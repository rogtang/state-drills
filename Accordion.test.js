import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

describe(`Accordion Component`, () => {
    const sectionsProp = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ];
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty li given no props', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})