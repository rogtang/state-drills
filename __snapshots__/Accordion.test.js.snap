// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens the last section when multiple sections have been clicked 1`] = `
<ul
  className="accordion"
>
  <li
    className="accordion-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="accordion-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<ul
  className="accordion"
>
  <li
    className="accordion-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="accordion-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty li given no props 1`] = `
<ul
  className="accordion"
/>
`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul
  className="accordion"
>
  <li
    className="accordion-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="accordion-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
